### PR TITLE
Branch 1: 

### PR DESCRIPTION
**Pull request by Eddie023**

### CODE REVIEW

File: src/controllers/adminsController.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  adminService
+    .createAdmin(req.body)
+    .then(data => res.status(HttpStatus.CREATED).json({ data }))
+    .catch(err => next(err));
+});
+
+/**
+ * PUT /api/admin/:id
+ */
+router.put('/:id', findAdmin, adminValidator, (req, res, next) => {
```
Comment (sudhirt4) [inline src/controllers/adminsController.js]: What is the purpose of findAdmin middleware here ?

File: src/controllers/projectController.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import * as projectService from '../services/projectService';
+
+const router = Router();
+
+/**
+ * GET /api/admins
+ */
+router.get('/', (req, res, next) => {
+  projectService
+    .getRelatedProject(req.headers)
```
Comment (sudhirt4) [inline src/controllers/projectController.js]: Services should be reusable anywhere in the project (even from cli). Don't tie it with headers directly.
Instead, pass the params required the function directly.

File: src/controllers/projectController.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+//     .then(data => res.json({ data }))
+//     .catch(err => next(err));
+// });
+
+/**
+ * POST /api/admin
+ */
+router.post('/', (req, res, next) => {
+  projectService
+    .createNewProject(req.body)
```
Comment (sudhirt4) [inline src/controllers/projectController.js]: Same as above

File: src/controllers/projectInstanceController.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    .then(data => res.json({ data }))
+    .catch(err => next(err));
+});
+
+/**
+ * POST /api/projectInstance
+ */
+router.post('/', (req, res, next) => {
+  projectInstanceService
+    .createProjectInstance(req.body)
```
Comment (sudhirt4) [inline src/controllers/projectInstanceController.js]: Same as above

File: src/controllers/signUpController.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+const router = Router();
+
+/**
+ * POST /api/signUp
+ */
+
+// create new admin
+router.post('/', (req, res, next) => {
+  adminService
+    .createAdmin(req.body)
```
Comment (sudhirt4) [inline src/controllers/signUpController.js]: Same as above

File: src/models/admin_project.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import bookshelf from '../db';
+
+const TABLE_NAME = 'admin_project';
```
Comment (sudhirt4) [inline src/models/admin_project.js]: What is the purpose of this table ?

File: src/migrations/20180803001052_admin_project.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
+/**
+ * @param  {object} knex
+ * @return {Promise}
+ */
+export function up(knex) {
+  return knex.schema.createTable('admin_project', table => {
```
Comment (sudhirt4) [inline src/migrations/20180803001052_admin_project.js]: Be consistent. Use either plural or singular for table names. We usually use plural.

File: src/models/projects.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import bookshelf from '../db';
+import ProjectInstance from './project_instances';
+
+const TABLE_NAME = 'projects';
+
+/* projects model*/
+
+class projects extends bookshelf.Model {
```
Comment (sudhirt4) [inline src/models/projects.js]: Use PascalCase

File: src/seeds/0001insert_project.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+export function seed(knex, Promise) {
+  // Deletes all existing entries
+  return knex('projects')
+    .del()
+    .then(() => {
+      // Inserts seed entries
+      return Promise.all([
+        knex('projects').insert([
+          {
+            project_name: 'Sentry',
```
Comment (sudhirt4) [inline src/seeds/0001insert_project.js]: Use prettier

File: src/services/projectService.js [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+    .fetchAll()
+    .then(data => {
+      const result = data.toJSON();
+
+      return result;
+    });
+
+  const projectId = [];
+
+  projects.forEach(element => {
```
Comment (sudhirt4) [inline src/services/projectService.js]: Use .map function for this purpose

Comment (sudhirt4) [reply src/services/projectService.js]: Actually, avoid this approach as a whole. Try to filter the projects at database level itself using joins

File: src/controllers/adminsController.js (hunk 1 above)
Comment (cyborge09) [reply src/controllers/adminsController.js]: to check the existence of the admin in the database, to update the admin credentials

File: src/models/projects.js (hunk 8 above)
Comment (cyborge09) [reply src/models/projects.js]: resolved

**Pull request merged**